enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,445 @@
+* As a MicroShift admin, I want to add additional interfaces to certain Pods so that I can
+  access them directly (from outside the cluster) without using Kubernetes' networking such as
+  NodePorts, Load Balancers, Ingresses, etc.
+* As a MicroShift admin, I want to add additional interfaces to certain Pods so that I can
+  start slowly migrating an existing solution to MicroShift.
+
+### Goals
+
+- Provide optional Multus CNI for MicroShift clusters that can be added to existing clusters
+
+### Non-Goals
+
+- Automatically removing Multus from the cluster upon RPM uninstall
+- Support Multus for multi-node deployments of MicroShift
+- Providing network policies or admission webhook for Multus
+
+## Proposal
+
+Deliver Multus for MicroShift as an optional RPM containing required manifests that will be applied
+during MicroShift's start. There should be little to no changes to MicroShift itself as we want
+Multus CNI for MicroShift to fit optional components pattern.
+
+Manifests for deploying Multus on MicroShift will be based on existing manifests for OpenShift,
+but they will differ because OpenShift uses thick architecture Multus whereas MicroShift will use
+thin architecture Multus.
+
+How should cleanup of the Multus artifacts look like is an open question (see Open Questions section).
+
+### Workflow Description
+
+**User** is a human user responsible for setting up and managing Edge Devices.
+**Application** is user's workload that intends to use additional interfaces.
+
+1. User gathers all information about the networking environment of the edge device.
+1. User prepares NetworkAttachmentDefinition (NAD) CRs that will allow Application to be part of the specified network.
+1. User prepares ostree commit that contains:
+   - (optional) Init procedures to configure OS for usage of the additional network
+   - MicroShift RPMs
+   - Multus for MicroShift RPM
+   - Application with the NetworkAttachmentDefinition CRs.
+1. User deploys the ostree commit onto the edge device.
+1. Edge device boots:
+1. (optional) Init procedures are configuring OS and networks
+1. MicroShift starts
+1. MicroShift applies Multus' manifests
+1. MicroShift applies Application's manifests that include NetworkAttachmentDefinitions
+1. Application's Pod are created, Multus inspects Pod's annotations and sets up 
+   additional interfaces based on matching NetworkAttachmentDefinitions
+1. Application's containers are running, they can utilize additional interfaces
+
+### API Extensions
+
+Multus is an established project with already existing API extensions.
+Following paragraphs does not present brand new CRDs or APIs, it only aims to summarize how adding
+Multus will affect MicroShift's API. For more information see
+[Multus CNI repository](https://github.com/openshift/multus-cni/).
+
+Multus is configured in following ways:
+- CNI configuration created when Multus' DaemonSet starts, it includes the primary CNI (ovn-kubernetes for MicroShift)
```

Comment:
This implies that any other pods that use CNI network need to wait until multus pod is ready, which may result in slightly increased pod startup latency (for pods such as dns). Is it something we want to mention in drawback section?

Reply:
Done